ui/tests/acceptance/allocation-detail-test.js

```diff
@@ -215,6 +216,42 @@ module('Acceptance | allocation detail', function(hooks) {
   });
 });
 
+// This is a separate module only because of the watchInTesting adjustment, hopefully that can change
+module('Acceptance | allocation detail after stopping', function(hooks) {
+  setupApplicationTest(hooks);
+  setupMirage(hooks);
+
+  hooks.beforeEach(async function() {
+    server.create('agent');
+
+    node = server.create('node');
+    job = server.create('job', { groupsCount: 1, createAllocations: false });
+    allocation = server.create('allocation', 'withTaskWithPorts', { clientStatus: 'running' });
+
+    this.owner.lookup('service:watch-list').set('watchInTesting', true);
+    await Allocation.visit({ id: allocation.id });
+  });
+
+  test('when the allocation endpoint 404s, a flash message appears', async function(assert) {
+    assert.equal(PageLayout.flashMessages.length, 0);
+
+    const controller = this.owner.lookup('controller:allocations/allocation');
+    const route = this.owner.lookup('route:allocations/allocation');
+    controller.watcher = route.watch.perform(controller.model, { throttle: 0, runInTests: true });
```

Comment:
As you know, these are distinctly not acceptance tests since they invoke methods on objects rather than treating the app like a black box.

I'm fine with running with this as-is and adding an overarching acceptance testing + watching task to the backlog.

To ramble for a minute...

I think the best solution will make use of either [ember-concurrency-test-waiter](https://github.com/bendemboski/ember-concurrency-test-waiter) or some abstraction over `run.cancelTimers` like I experimented with over here: https://github.com/hashicorp/nomad/commit/d5e41177c9dcd69d079230403a46f49e78b95fed

The dream is to have a test that is easy to write, reads well, and works via a series of button clicks and waiting. Which is a tall order given the inner workings of Ember concurrency.